encord/orm/dataset.py

```diff
@@ -120,14 +211,180 @@ def created_at(self, value: datetime) -> None:
+
+    def get_client_metadata(self):
+        """
+        This method lazily retrieves custom client metadata and cache it.
+        """
+        return self.client_metadata
+
+    @property
+    def width(self) -> int:
+        return self["width"]
+
+    @width.setter
+    def width(self, new_width: int) -> None:
+        self["width"] = new_width
+
+    @property
+    def height(self) -> int:
+        return self["height"]
+
+    @height.setter
+    def height(self, new_height: int) -> None:
+        self["height"] = new_height
+
+    @property
+    def last_edited_at(self) -> datetime:
+        return parser.parse(self["last_edited_at"])
+
+    @last_edited_at.setter
+    def last_edited_at(self, new_last_edited_at: datetime) -> None:
+        self["last_edited_at"] = new_last_edited_at.strftime(DATETIME_STRING_FORMAT)
+
+    @property
+    def file_link(self) -> str:
+        return self["file_link"]
+
+    @file_link.setter
+    def file_link(self, new_file_link: str) -> None:
+        self["file_link"] = new_file_link
+
+    @property
+    def signed_url(self) -> str:
+        if self.data_type in [DataType.VIDEO, DataType.IMG_GROUP, DataType.IMAGE] and self["signed_url"] is None:
+            if self.querier is not None:
+                payload = {"data_type": self.data_type.value}
+                res = self.querier.basic_getter(SignedUrl, uid=self.uid, payload=payload)
+                self["signed_url"] = res.signed_url
+            else:
+                raise EncordException(f"Could not get signed url for DataRow with uid: {self.uid}")
+        return self["signed_url"]
+
+    @signed_url.setter
+    def signed_url(self, new_signed_url: str) -> None:
+        self["signed_url"] = new_signed_url
+
+    def get_signed_url(self):
+        """
+        This method lazily generates and returns cached signed url for the given DataRow.
+        """
+        return self.signed_url
+
+    @property
+    def file_size(self) -> int:
+        return self["file_size"]
+
+    @file_size.setter
+    def file_size(self, new_file_size: int) -> None:
+        self["file_size"] = new_file_size
+
+    @property
+    def file_type(self) -> str:
+        return self["file_type"]
+
+    @file_type.setter
+    def file_type(self, new_file_type: str) -> None:
+        self["file_type"] = new_file_type
+
+    @property
+    def storage_location(self) -> StorageLocation:
+        return self["storage_location"]
+
+    @storage_location.setter
+    def storage_location(self, new_storage_location: StorageLocation) -> None:
+        self["storage_location"] = new_storage_location
+
+    def get_images(self, get_signed_url: bool = False) -> List[ImageData]:
+        """
+        This method lazily retrieves and returns cached list of images for the given DataRow image group.
+        If the given DataRow is not an image group then this method simply returns None.
```

Comment:
```suggestion
        If the given DataRow is not an image group then this method simply returns `None`.
```
Let's always specify back ticks for Python code.